### PR TITLE
content key policy exists

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Media Services Content Key Policy Exists',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:5]:
```suggestion
    title: 'Media Services Content Key Policy',
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    category: 'Media Services',
+    domain: 'Content Delivery',
+    description: 'Ensure that Microsoft Azure Media Services have Content Key Policy configured.',
+    more_info: 'A Content Key Policy in Azure Media Services dictates how content keys, ensuring secure asset access, are delivered to end clients. It allows setting requirements or restrictions that keys with specific configurations must meet before being delivered to clients.',
+    link: 'https://learn.microsoft.com/en-us/azure/media-services/latest/drm-content-key-policy-concept',
+    recommended_action: 'Modify media service account and add a content key policy.',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:11]:
```suggestion
    recommended_action: 'Modify media service account and add content key policy.',
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+            for (let mediaService of mediaServices.data) {
+                if (!mediaService.id) continue;
+
+                var listContentKeyPolicies = helpers.addSource(cache, source, 
+                    ['mediaServices', 'listContentKeyPolicies', location, mediaService.id]);
+
+                if (!listContentKeyPolicies || listContentKeyPolicies.err || !listContentKeyPolicies.data) {
+                    helpers.addResult(results, 3, `Unable to query for Content Key Policy: ${helpers.addError(listContentKeyPolicies)}`,
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:42]:
```suggestion
                    helpers.addResult(results, 3, `Unable to query Content Key Policy for Media service account: ${helpers.addError(listContentKeyPolicies)}`,
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                        location, mediaService.id);
+                    continue;
+                }
+                if (listContentKeyPolicies.data.length > 0) {
+                    helpers.addResult(results, 0, 'Azure Media Service has content key policy configured', location, mediaService.id);
+                } else {
+                    helpers.addResult(results, 2, 'Azure Media Service does not have content key policy configured', location, mediaService.id);
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:49]:
```suggestion
                    helpers.addResult(results, 2, 'Media Service account does not have content key policy configured', location, mediaService.id);
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.spec.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                done();
+            });
+        });
+
+        it('should give unknown result if unable to get media service', function(done) {
+            const cache = createCache([mediaServices[0]], null);
+            amsContentKeyPolicy.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].message).to.include('Unable to query for Content Key Policy');
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.spec.js:80]:
```suggestion
                expect(results[0].message).to.include('Unable to query Content Key Policy for Media service account:');
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.spec.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                done();
+            });
+        });
+
+        it('should give passing result if content key policy exist', function(done) {
+            const cache = createCache([mediaServices[0]], [listContentKeyPolicies[1]]);
+            amsContentKeyPolicy.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Azure Media Service has content key policy configured');
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.spec.js:91]:
```suggestion
                expect(results[0].message).to.include('Media Service account has content key policy configured');
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.spec.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                done();
+            });
+        });
+
+        it('should give failing result if content key policy not exist', function(done) {
+            const cache = createCache([mediaServices[0]], []);
+            amsContentKeyPolicy.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Azure Media Service does not have content key policy configured');
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.spec.js:102]:
```suggestion
                expect(results[0].message).to.include('Media Service account does not have content key policy configured');
```

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Media Services Content Key Policy',
+    category: 'Media Services',
+    domain: 'Content Delivery',
+    description: 'Ensure that Azure Media Services have Content Key Policy configured.',
+    more_info: 'A Content Key Policy in Azure Media Services dictates how content keys, ensuring secure asset access, are delivered to end clients. It allows setting requirements or restrictions that keys with specific configurations must meet before being delivered to clients.',
+    link: 'https://learn.microsoft.com/en-us/azure/media-services/latest/drm-content-key-policy-concept',
+    recommended_action: 'Modify media service account and add content key policy.',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:11]:
```suggestion
    recommended_action: 'Modify Media Service account and add content key policy.',
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/azure/mediaServices/amsContentKeyPolicy.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+                var listContentKeyPolicies = helpers.addSource(cache, source, 
+                    ['mediaServices', 'listContentKeyPolicies', location, mediaService.id]);
+
+                if (!listContentKeyPolicies || listContentKeyPolicies.err || !listContentKeyPolicies.data) {
+                    helpers.addResult(results, 3, `Unable to query Content Key Policy for Media service account: ${helpers.addError(listContentKeyPolicies)}`,
+                        location, mediaService.id);
+                    continue;
+                }
+                if (listContentKeyPolicies.data.length > 0) {
```
Comment (mehakseedat63) [inline plugins/azure/mediaServices/amsContentKeyPolicy.js:46]:
```suggestion
                if (listContentKeyPolicies.data.length) {
```
don't need to check for > 0

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**